pymeasure/instruments/teledyne/teledyneT3AFG.py

```diff
@@ -0,0 +1,131 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+from pymeasure.instruments import Instrument, Channel
+from pymeasure.instruments.validators import strict_range, strict_discrete_set
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class SignalChannel(Channel):
+
+    def get_process_generator(keyword, unit, type):
+        """Generate a get_process method searching for keyword, stripping unit"""
+        def selector(values):
+            if keyword in values:
+                try:
+                    return type(values[values.index(keyword)+1].strip(unit))
+                except (ValueError, IndexError):
+                    # Something went quite wrong if the keyword exists but the value doesn't
+                    return None
+            else:
+                # Wrong wavetype for this keyword
+                return None
+        return selector
```

Comment:
I chose to put the util method outside the channel class in this case because I can imagine it being useful for the larger instrument class if someone wanted to implement instrument level controls